### ISSUE TITLE
[question] deps_cpp_info build_paths confusion

### ISSUE THREAD

**Issue by keithrob91** [conan-io/conan#3956]
Conan 1.9.1

The documentation for build_paths states: 

> Same as builddirs but transformed to absolute paths

And builddirs states:

> Ordered list with build scripts paths, by default [‘’]. CMake will search in these dirs for cmake files, like findXXX.cmake

However, I am seeing `self.deps_cpp_info[some-pkg-name].build_paths[0]` return the package directory for 'some-pkg-name' i.e. same as `rootpath`, but with an added '/' on the end, even though 'some-pkg-name' put nothing into `builddirs`.

Is this expected?

**Comment by jgsogo** [comment]
It seems a bug. According to the docs it makes no sense to have `build_paths` if no `builddirs` are defined. It should be easy to fix.

Thanks for the report.

**Comment by danimtb** [comment]
Hi @keithrob91 and thanks a lot for reporting.

It seems this was a bit misleading in the docs.  ``cpp_info.builddirs`` store relative paths to directories containing build scripts and the base path used for them is the ``self.package_folder``. The default value for ``cpp_info.builddirs`` is ``[""]``, that means it it actually the ``self.package_folder``. That is de reason you are seeing the full package folder path in ``deps_cpp_info["package-name"].build_paths``, so this is the expected behavior.

To sum up:
If package "A" declares a `self.cpp_info.builddirs = ["my_build_dir"]`, a consumer of this package will get ``self.deps_cpp_info["A"].build_paths`` with value ``["/home/user/.conan/A/version/user/channel/package/e2198uoijdwqljfd/my_build_dir/"]``

I have submitted a PR to the documentation to improve the description: https://github.com/conan-io/docs/pull/952

Hope that makes sense. Thanks!
